lib/ht_sip_validator/validation/image/filenames.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module HathiTrust::Validation::Image
+  class Filenames < HathiTrust::Validation::Base
+
+
+    def perform_validation
+      image_files = HathiTrust::Validation::Image.image_files(@sip.files)
+      return no_images_error if image_files.empty?
+
+      bad_filenames = image_files.select{|filename| is_valid_filename? filename}
+      bad_filenames.map do |filename|
+        create_error(
+          validation: :image_filename,
+          human_message: "Base filename of #{filename} is not numeric only.",
+          extras: { filename: filename }
+        )
+      end
+    end
+
+    private
+    # Convenience method to create the error in the event of an error
+    # which obviates the rest of the logic
+    def no_images_error
+        create_error(
+          validation: :image_sequence,
+          human_message: "No image filenames recognized.",
+          extras: { image_count: 0 }
+        )
+    end
+
+    def is_valid_filename?(filename)
+      File.basename(filename) =~ /^\d+$/
```

Comment:
Crap.  I've coded myself into a corner.  In order to even be in the list of image_files, the extension is checked against a constant defined at the module level.
```HathiTrust::Validation::Image::IMG_FILE_EXTENSIONS = [".jp2", ".tif", ".png"].freeze```
However, that means if the extensions are checked again elsewhere, both of those checks need to be kept in sync.  Maybe it's just a bad function name.  Perhaps ``is_basename_valid?``` would be more fitting.

I assumed png where in there since they're specified in the HathiTrust Data API.  I'll remove png from that array.